src/store/registry.js

```diff
@@ -10,20 +11,20 @@ export class StoreRegistry {
     this.defaultScope = defaultScope;
   }
 
-  initStore = (key, Store, fromContainer) => {
+  initStore = (key, Store, config) => {
     const { initialState, actions } = Store;
 
-    if (Store.containedBy && !fromContainer) {
-      Promise.reject(
-        new Error(
-          `Store ${Store.key} should be contained by a container but it is used globally. ` +
-            `While it might still work, it will likely cause unexpected behaviours.`
-        )
+    if (Store.containedBy && !config.contained(Store)) {
```

Comment:
It is not apparent to me why we pass in a predicate while we can already check the result at the call site?

Reply:
True, right now it is superfluous. I ported it from #201 where Store can be a different one but maybe we can evaluate at call site even there. Will check and eventually simplify in that PR.